### DISCUSSION TITLE
SubUrl (sub directory) - does it work?

### DISCUSSION THREAD

**Discussion by alexkoon** [healthchecks/healthchecks#900] [Q&A] [answered]
I am trying to run healthchecks.io on a sub url (https://www.mydomain.com/healthchecks for example) but failing to get this to work. I am running this behind Treafik reverse proxy, and I can get it to strip the subURL from the request before forwarding to healthchecks.io.

The screen does not render properly and looking at the HTML, I can see lots of relative path references to '/static/xxxxx'.

I cannot find anything under the configuration settings that seems to help this situation and none of the historical issues seem to have a solution. Have I missed something obvious?

**Comment by cuu508** [comment, marked as answer]
Serving under a subpath is not well tested – it may be possible to get to work, but you may run into issues (PRs welcome). There is also no specific documentation to get it to work.

I just tested the following setup: 

On my local system, I started a development web server with the following additional settings in `local_settings.py`:

```python
FORCE_SCRIPT_NAME = "/foo"
STATIC_URL = "/foo/static/"
CSRF_TRUSTED_ORIGINS = ["http://localhost"]
```

It serves on http://localhost:8000. 

On the same system I configured nginx like so:

```
user www-data;
worker_processes auto;
worker_cpu_affinity auto;
pid /run/nginx.pid;

events {
    worker_connections 4000;
    use epoll;
    accept_mutex off;
}

http {
    include /etc/nginx/mime.types;
    default_type application/octet-stream;
    error_log /var/log/nginx/error.log;

    server {
        listen 80;
        location /foo {
            proxy_pass http://127.0.0.1:8000/;
        }

        location /foo/static {
            expires max;
            add_header Cache-Control "public";
            alias /fixme/path/to/my/healthchecks/static-collected;
            gzip_static on;
        }

    }

}
```

After reloading nginx configuration, I can visit `http://localhost/foo` and the site seems to nominally work: I see the login form, CSS loads, I can log in, view check's details page etc. I have not tested every corner of the site and it's quite possible something somewhere is broken too.

**Reply by alexkoon** [reply]
Thanks for the concise reply. I gave it a spin mapping your nginx config to traefik. On the surface it looks much better. Being a new user,  I have reverted to using a sub-domain for now to test out.
